pkg/hostagent/packet_event.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hostagent
+
+import (
+	"encoding/json"
+	"net"
+	"io"
+	"os"
+	"reflect"
+)
+
+type PacketEvent struct {
+    TimeStamp string
+    DropReason string
+    SourceMac string
+    DestinationMac string
+    EtherType string
+    SourceIP string
+    DestinationIP string
+    IPProto string
+    SourcePort string
+    DestinationPort string
+}
+
+func (agent *HostAgent) RunPacketEventListener(stopCh <-chan struct{}) {
+    os.Remove(agent.config.PacketEventNotificationSock)
+    pc, err := net.Listen("unix", agent.config.PacketEventNotificationSock)
+    if err != nil {
+        agent.log.Errorf("Failed to listen on unix socket %s: %s ",
+               agent.config.PacketEventNotificationSock, err)
+        return
+    } else {
+        agent.log.Info("Listening for packet events on unix socket ",
+			agent.config.PacketEventNotificationSock)
+    }
+
+    go func() {
+        for {
+            fd, err := pc.Accept()
+            if err != nil {
+		agent.log.Warnf("Failed to accept %s\n", err)
+                return
+            }
+            go func() {
+                buffer := make([]byte, 4096)
+                n, err := fd.Read(buffer)
+                if err != nil {
+                    if err != io.EOF {
+                        agent.log.Errorf("packet event socket read error %s\n", err)
+                    }
+                    fd.Close()
+                    return
+                }
+                var m [30]PacketEvent
+                err1 := json.Unmarshal(buffer[:n], &m)
+                agent.log.Infof("PacketEvent Received length %d \n",n)
+                if err1 != nil {
+			agent.log.Info("Unmarshaling error ", err1)
+                }
+		numRecords := n/int(reflect.TypeOf(m[0]).Size())
```

Comment:
This would be a good place to get time.Now()

Reply:
Sure. Am passing it as an argument now